ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -927,12 +928,36 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{app
                 title: <ActionMenuItem actionLabel='Diff' />,
                 action: () => this.selectNode(fullName, 0, 'diff'),
                 disabled: app.status.sync.status === appModels.SyncStatuses.Synced
-            },
-            {
+            }
+        ];
+
+        if (!app.status.operationState?.finishedAt || app.status.operationState?.phase === 'Running') {
```

Comment:
I _think_ this might be the first time that a button in this toolbar will appear and disappear while in the view, which might seem strange to some users. It's probably ok for now. And if there are complaints, we can address it in another PR.